### PR TITLE
[MAINTENANCE] Reduce output amount in Jupyter notebooks when displaying DataAssistantResult

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
By adjusting `DataAssistantResult.__repr()__` to return only summary information when rendered in a `Jupyter` notebook, the notebook experience is made more tractable.  The complete `DataAssistantResult` information is still available using `print()` and accessor methods.  Using print(result) returns full details.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-10142
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py [hunk 1]
```diff
@@ -281,7 +278,7 @@ def _add_onboarding_data_assistant_cells(self) -> None:
         self._add_onboarding_data_assistant_instructions()
         self.add_code_cell(
             code="""\
-data_assistant_result: DataAssistantResult = context.assistants.onboarding.run(
+result = context.assistants.onboarding.run(
```
Comment (cdkini) [inline great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py:281]: Why lose the type hints?

File: great_expectations/core/expectation_validation_result.py [hunk 2]
```diff
@@ -147,15 +147,14 @@ def __ne__(self, other):
             # if invalid comparisons are attempted, the objects are not equal.
             return True
 
-    def __repr__(self):
+    def __repr__(self) -> str:
```
Comment (cdkini) [inline great_expectations/core/expectation_validation_result.py:150]: Could we adjust the thresholds for type hinting?

File: great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py:281]: @cdkini Type hints are not favored in notebooks (we have followed this policy before, but recently deviated by mistake).  Thanks.

File: great_expectations/core/expectation_validation_result.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/core/expectation_validation_result.py:150]: @cdkini Not sure what to do here -- please advise.  Thanks.

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py [hunk 3]
```diff
@@ -97,6 +104,10 @@ class DataAssistantResult(SerializableDictDot):
         "usage_statistics_handler",
     }
 
+    IN_JUPYTER_NOTEBOOK_KEYS = {
+        "execution_time",
+    }
+
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py:110]: Could we add a comment that let's us know what this is for?

File: great_expectations/core/expectation_validation_result.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/core/expectation_validation_result.py:150]: @cdkini I think that "dunder" methods should be added to exclusion list (just like private methods) -- is it possible?  Thanks.

**Review by cdkini — APPROVED** [review]

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py:110]: @cdkini I can add a comment (though the constant's name accurate describes it too).  Thanks.

File: great_expectations/core/expectation_validation_result.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/core/expectation_validation_result.py:150]: Update in `scripts/check_type_hint_coverage.py`

Comment (cdkini) [reply great_expectations/core/expectation_validation_result.py:150]: And dunder methods are included in `mypy` outputs so we want to be consistent with that tool. Not a big deal but it helps us keep our code a bit more stringent.

**Review by NathanFarmer — APPROVED** [review]
UAT performed and ✅

**Pull request merged**